mindsdb_native/libs/phases/data_analyzer/data_analyzer.py

```diff
@@ -299,18 +299,22 @@ def run(self, input_data):
                     stats_v2[col_name]['bias']['warning'] = warning_str + " This doesn't necessarily mean there's an issue with your data, it just indicates a higher than usual probability there might be some issue."
 
                 if data_type == DATA_TYPES.NUMERIC:
-                        outliers = lof_outliers(data_subtype, col_data)
-                        stats_v2[col_name]['outliers'] = {
-                            'outlier_values': outliers,
-                            'outlier_buckets': compute_outlier_buckets(
-                                outlier_values=outliers,
-                                hist_x=histogram['x'],
-                                hist_y=histogram['y'],
-                                percentage_buckets=percentage_buckets,
-                                col_stats=stats_v2[col_name]
-                            ),
-                            'description': """Potential outliers can be thought as the "extremes", i.e., data points that are far from the center of mass (mean/median/interquartile range) of the data."""
-                        }
+                    outliers = lof_outliers(data_subtype, col_data)
+                    stats_v2[col_name]['outliers'] = {
+                        'outlier_values': outliers,
+                        'outlier_buckets': compute_outlier_buckets(
+                            outlier_values=outliers,
+                            hist_x=histogram['x'],
+                            hist_y=histogram['y'],
+                            percentage_buckets=percentage_buckets,
+                            col_stats=stats_v2[col_name]
+                        ),
+                        'description': """Potential outliers can be thought as the "extremes", i.e., data points that are far from the center of mass (mean/median/interquartile range) of the data."""
+                    }
+                    # specify positive numerical domain
+                    if stats_v2[col_name]['histogram']['x'][0] >= 0:
```

Comment:
is it assumed that `histogram['x']` is sorted?

Reply:
It does. This is only for the numerical data type, and from what I can gather, `np.histogram` returns sorted bins.